src/lib/starRelease.js

```diff
@@ -0,0 +1,86 @@
+import { linearSR, conditionalSR } from 'azimuth-js';
+import { Just, Nothing } from 'folktale/maybe';
+
+function mapRange(amount, cb) {
+  return [...Array(amount).keys()].map(cb);
+}
+
+// Preferentially withdraw from conditional contracts and use linear
+// to mop up the rest
+export function generateWithdrawTxs(contracts, batchLimits, amount, to) {
```

Comment:
wh-what

I don't understand at all why we're mixing the use of linear and conditional, both here and in `lib/useStarReleaseStore`. If this is somehow useful for creating the desired UI flows, I'd expect that to happen in those UI flows, not in a lib...

Reply:
I put in the lib because I didn't want the UI code to become too cluttered. I thought it would be cleaner to hide the difference between the contracts in a lib, because from the bridge perspective, the two contracts don't really need to be exposed differently.

Reply:
That sounds like fair game. Carry on!